test/e2e/aks.go

```diff
@@ -303,3 +310,48 @@ func GetWorkingAKSKubernetesVersion(ctx context.Context, subscriptionID, locatio
 
 	return maxVersion, nil
 }
+
+// GetLatestStableAKSKubernetesVersion returns the latest stable available Kubernetes version of AKS.
+func GetLatestStableAKSKubernetesVersion(ctx context.Context, subscriptionID, location, version string) (string, error) {
+	settings, err := auth.GetSettingsFromEnvironment()
+	if err != nil {
+		return "", err
+	}
+	authorizer, err := settings.GetAuthorizer()
+	if err != nil {
+		return "", err
+	}
+	containerServiceClient := containerservice.NewContainerServicesClient(subscriptionID)
+	containerServiceClient.Authorizer = authorizer
+	result, err := containerServiceClient.ListOrchestrators(ctx, location, ManagedClustersResourceType)
+	if err != nil {
+		return "", err
+	}
+
+	var orchestratorversions []string
+	var isaksversioninpreview []*bool
+	var foundWorkingVersion bool
+	var orchVersion string
+	var maxVersion string
+
+	for _, o := range *result.Orchestrators {
+		orchestratorversions = append(orchestratorversions, *o.OrchestratorVersion)
+		isaksversioninpreview = append(isaksversioninpreview, o.IsPreview)
+	}
+
+	for i := range isaksversioninpreview {
+		if isaksversioninpreview[i] != nil {
+			orchVersion = orchestratorversions[i-1]
```

Comment:
why i-1 and not i here? shouldn't these match?

also this whole logic relies on the ordering returned by AKS, right? I would suggest actually parsing the resulting versions as semver, keep a pointer to the maximum, and do semver compare against each version, updating the max as you go.

Reply:
ah, i-1 because the last versions would be preview in an ordered list. so we're finding the version preview version in the array and then taking one item back from the orchestrator version list, assuming that won't be a preview version. 

I would suggest the approach above, which will be much clearer to read in the future as well. I don't believe we guarantee the ordering you're relying on.

Reply:
Thank you for your comments. 
I have made the change and used semver.Sort to find the maximum version.